src/lib/NumberFormatting.ts

```diff
@@ -45,3 +45,47 @@ export function calculateDisplayedDecimals(amount: number | null, currency: Cryp
     if (amount < 10 * 1e5) return Math.max(decimals.value, 1);
     return decimals.value;
 }
+
+export function numberToLiteral(n: number): string {
+    // https://stackoverflow.com/questions/5529934/javascript-numbers-to-words
+    const ones = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
+    const tens = ['', '', 'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety'];
+    const teens = ['ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen',
+        'sixteen', 'seventeen', 'eighteen', 'nineteen'];
+
+    function convertMillions(num: number): string {
+        if (num >= 1000000) {
+            return `${convertMillions(Math.floor(num / 1000000))} million ${convertThousands(num % 1000000)}`;
+        }
+        return convertThousands(num);
+    }
+
+    function convertThousands(num: number): string {
+        if (num >= 1000) return `${convertHundreds(Math.floor(num / 1000))} thousand ${convertHundreds(num % 1000)}`;
+        return convertHundreds(num);
+    }
+
+    function convertHundreds(num: number): string {
+        if (num > 99) return `${ones[Math.floor(num / 100)]} hundred ${convertTens(num % 100)}`;
+        return convertTens(num);
+    }
+
+    function convertTens(num: number): string {
+        if (num < 10) return ones[num];
+        if (num >= 10 && num < 20) return teens[num - 10];
+        return `${tens[Math.floor(num / 10)]} ${ones[num % 10]}`;
+    }
+
+    if (n === 0) return 'zero';
+    return convertMillions(n);
+}
+
+export function numberToLiteralTimes(n: number): string {
+    const timesTable = ['', 'once', 'twice', 'thrice'];
+
+    n = parseInt(n.toString(), 10);
+    if (n <= 0) throw new Error('Invalid Input! Times number must be positive >= 1!');
+    if (n < timesTable.length) return timesTable[n];
+
+    return `${numberToLiteral(n)} times`;
+}
```

Comment:
This is missing internationalization.
You can import `i18n` from `i18n-setup.ts` and then use its `t` method.
Using variables is also supported in source strings, which shall be useful here, e.g.:
```
i18n.t('{millions} million {remainder}', {
    millions: convertMillions(Math.floor(num / 1000000)),
    remainder: convertThousands(num % 1000000),
})
```